### PR TITLE
feat: use `pytest-databases` and updated `advanced-alchemy`

### PR DESCRIPTION
**Pull request by cofin**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

- 

### Close Issue(s)

-

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -91,7 +91,15 @@ linting = [
...
   "pylint",
 ]
-test = ["pytest", "pytest-xdist", "pytest-mock", "pytest-cov", "coverage", "pytest-sugar"]
+test = [
+  "pytest",
+  "pytest-xdist",
+  "pytest-mock",
+  "pytest-cov",
+  "coverage",
+  "pytest-sugar",
```
Comment (JacobCoffee) [inline pyproject.toml:100]: we should add this everywhere

Comment (cofin) [reply pyproject.toml:100]: yep.  I agree.  The output is much cleaner.

File: pyproject.toml [hunk 2]
```diff
@@ -45,7 +45,7 @@ app = "app.__main__:run_cli"
 
 [build-system]
 build-backend = "hatchling.build"
-requires = ["hatchling", "nodeenv"]
+requires = ["hatchling", "nodeenv", "setuptools"]
```
Comment (JacobCoffee) [inline pyproject.toml:48]: why setuptools?

Comment (cofin) [reply pyproject.toml:48]: Actually, this can be removed.  This was to debug an issue in my local env.

Comment (cofin) [reply pyproject.toml:48]:
```suggestion
requires = ["hatchling", "nodeenv"]
```

**Review by JacobCoffee — APPROVED** [review]
New Scalar UI doesnt support password flow but will soon https://github.com/scalar/scalar/discussions/1389

**Pull request merged**